spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/AuthConf.scala

```diff
@@ -35,7 +35,23 @@ case class PasswordAuthConf(user: String, password: String) extends AuthConf {
 
 /** Obtains authentication configuration by reading  `SparkConf` object. */
 trait AuthConfFactory {
+
   def authConf(conf: SparkConf): AuthConf
+
+  /** List of extra properties allowed in SparkConf passed to `authConf` method */
+  def properties: Set[String] = Set.empty
+}
+
+object AuthConfFactory {
+  val AuthConfFactoryProperty = "spark.cassandra.auth.conf.factory"
+  val Properties = Set(AuthConfFactoryProperty)
```

Comment:
true, i am thinking in terms of a standard Akka class of settings which use final val MyVal syntax vs an object